snuba/datasets/entity.py

```diff
@@ -149,3 +150,13 @@ def describe(self) -> Description:
                 Description(header="Relationships", content=relationships),
             ],
         )
+
+    @staticmethod
+    def get_extra_conditions_for_subscriptions_query(
```

Comment:
You're messing up bounded contexts here. Entities are meant to not have any knowledge of how they're used, subscriptions vs. queries etc. So having subscription specific functions in the entity is not correct. Subscription logic should stay in the subscription code.